typescript/cli/src/deploy/warp.ts

```diff
@@ -128,10 +151,110 @@ async function executeDeploy(params: DeployParams) {
     log('Writing deployment artifacts');
     await registry.addWarpRoute(warpCoreConfig);
   }
-  log(JSON.stringify(warpCoreConfig, null, 2));
+  log(indentYamlOrJson(yamlStringify(warpCoreConfig, null, 2), 4));
   logBlue('Deployment is complete!');
 }
 
+async function deployAndResolveWarpIsm(
+  warpConfig: WarpRouteDeployConfig,
+  multiProvider: MultiProvider,
+  registry: IRegistry,
+  ismFactoryDeployer: HyperlaneProxyFactoryDeployer,
+): Promise<WarpRouteDeployConfig> {
+  return promiseObjAll(
+    objMap(warpConfig, async (chain, config) => {
+      // Skip deployment if Ism is empty, or a string
```

Comment:
what is the case where it would be a string? is that a bug or is this something we'll come back to?